src/main/java/com/exactpro/epfast/decoder/integer/DecodeMandatoryInt32.java

```diff
@@ -9,24 +9,58 @@
 
     private static final int NEGATIVE_LIMIT = Integer.MIN_VALUE >> 7;
 
+    private boolean checkPositiveSingExtension = false;
+
+    private boolean checkNegativeSingExtension = false;
+
     private int value;
 
     public void decode(ByteBuf buf) {
         reset();
+        checkPositiveSingExtension = false;
```

Comment:
you need only one flag. You can determine if it's positive or negative based on `this.value`